atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/HostIdResult.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra;
+
+import com.palantir.logsafe.Preconditions;
+import java.util.Set;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface HostIdResult {
+    enum Type {
+        // Failure should affect quorum calculations (network failure)
+        HARD_FAILURE,
+        // Failure shouldn't affect quorum calculations (i.e. missing API in C*)
+        SOFT_FAILURE,
+        SUCCESS
+    }
+
+    Type type();
+
+    Set<String> hostIds();
+
+    @Value.Check
+    default void checkHostIdsStateBasedOnResultType() {
```

Comment:
While I generally like having illegal states unrepresentable (ADTs), I think this is fine given the constructors.